libraries/lib-wx-wrappers/SpinControl.h

```diff
@@ -0,0 +1,84 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/**********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file SpinControl.h
+
+ Dmitry Vedenko
+
+ **********************************************************************/
+#pragma once
+
+#include <numeric>
+
+#include <wx/control.h>
+
+#include "TranslatableString.h"
+#include "Prefs.h"
+
+class wxTextCtrl;
+class wxSpinButton;
+class wxKeyEvent;
+
+class WX_WRAPPERS_API SpinControl final
+   : public wxControl
+   , private PrefsListener
+{
+public:
+   using ValueType = double;
+   
+   SpinControl(
+      wxWindow* parent, wxWindowID winid = wxID_ANY, ValueType value = 0.0,
+      ValueType min = 0.0f, ValueType max = 100.0, ValueType step = 1.0,
+      bool allowFractional = false, const wxPoint& pos = wxDefaultPosition,
+      const wxSize& size = wxDefaultSize, const TranslatableString& name = {});
```

Comment:
I might have used a chain-calling Options class for “named parameter idiom” for the ValueTypes and the bool, so the call sites might be easier.